docs/manual/manual_circuit.md

```diff
@@ -0,0 +1,1935 @@
+```{code-cell} ipython3
+
+from pytket import Circuit
+
+circ = Circuit(3)
+circ.CX(0, 1).CZ(1, 2).X(1).Rx(0.3, 0)
+
+print("Total gate count =", circ.n_gates)
+print("Circuit depth =", circ.depth())
+```
+
+As characteristics of a {py:class}`~pytket.circuit.Circuit` go, these are pretty basic. In terms of approximating the noise level, they fail heavily from weighting all gates evenly when, in fact, some will be much harder to implement than others. For example, in the NISQ era, we find that most technologies provide good single-qubit gate times and fidelities, with two-qubit gates being much slower and noisier [^cite_arut2019]. On the other hand, looking forward to the fault-tolerant regime we will expect Clifford gates to be very cheap but the magic $T$ gates to require expensive distillation procedures [^cite_brav2005] [^cite_brav2012].
+
+We can use the {py:class}`~pytket.circuit.OpType` enum class to look for the number of gates of a particular type. Additionally, the methods {py:meth}`~pytket.circuit.Circuit.n_1qb_gates`, {py:meth}`~pytket.circuit.Circuit.n_2qb_gates` and {py:meth}`~pytket.circuit.Circuit.n_nqb_gates` can be used to count the number of gates in terms of how many qubits they act upon irrespective of type.
+
+We also define $G$-depth (for a subset of gate types $G$) as the minimum number of layers of gates in $G$ required to run the {py:class}`~pytket.circuit.Circuit`, allowing for topological reorderings. Specific cases of this like $T$-depth and $CX$-depth are common to the literature on circuit simplification [^cite_amy2014] [^cite_meij2020].
+
+
+```{code-cell} ipython3
+
+from pytket import Circuit, OpType
+from pytket.circuit.display import render_circuit_jupyter as draw
+
+circ = Circuit(3)
+circ.T(0)
+circ.CX(0, 1)
+circ.CX(2, 0)
+circ.add_gate(OpType.CnRy, [0.6], [0, 1, 2])
+circ.T(2)
+circ.CZ(0, 1)
+circ.CZ(1, 2)
+circ.T(1)
+
+draw(circ) # draw circuit diagram
+
+print("T gate count =", circ.n_gates_of_type(OpType.T))
+print("#1qb gates =", circ.n_1qb_gates())
+print("#2qb gates =", circ.n_2qb_gates())
+print("#3qb gates =", circ.n_nqb_gates(3)) # count the single CnRy gate (n=3)
+print("T gate depth =", circ.depth_by_type(OpType.T))
+print("2qb gate depth =", circ.depth_by_type({OpType.CX, OpType.CZ}))
+```
+
+```{note}
+Each of these metrics will analyse the {py:class}`~pytket.circuit.Circuit` "as is", so they will consider each Box as a single unit rather than breaking it down into basic gates, nor will they perform any non-trivial gate commutations (those that don't just follow by deformation of the DAG) or gate decompositions (e.g. recognising that a $CZ$ gate would contribute 1 to $CX$-count in practice).
+```
+
+Its also possible to count all the occurrences of each {py:class}`~pytket.circuit.OpType` using the {py:func}`~pytket.utils.stats.gate_counts` function from the {py:mod}`pytket.utils` module.
+
+
+```{code-cell} ipython3
+
+    from pytket.utils.stats import gate_counts
+
+    gate_counts(circ)
+```
```

Comment:
I missed this one in the last review

Reply:
Thanks Will, fixed in [4b7448b](https://github.com/CQCL/pytket-docs/pull/370/commits/4b7448be05e9078e563f542668fafb25ca241328)

Reply:
I did a search to try and rule out other indentation issues, didn't find any biut maybe the search pattern could be wrong.